app/controllers/posts_controller.rb

```diff
@@ -9,6 +9,32 @@ def show
     @post = Post.find(params[:id])
   end
 
+  def new
+    @user = current_user
+    post = Post.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: } }
+    end
+  end
+
+  def create
+    @user = current_user
+    post = Post.new(params.require(:post).permit(:text, :title))
+    puts post.user_id = @user.id
+    respond_to do |format|
+      format.html do
+        if post.save
+          post.increment_counter
+          flash[:success] = 'Post created!'
+          redirect_to user_posts_path(@user)
+        else
+          flash.now[:error] = 'Post not created!'
+          render :new, locals: { post: }
+        end
+      end
+    end
+  end
```

Comment:
### **OPTIONAL**

- [ ] Please enhance your code with the [Strong Parameter](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.)